torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1360,7 +1363,7 @@ def cast_symbool_to_symint_guardless(symbool: torch.SymBool) -> torch.SymInt:
     'IsNonOverlappingAndDenseIndicator': eval_is_non_overlapping_and_dense,
     'floor': math.floor,
     'ceiling': math.ceil,
-    'cast_symbool_to_symint_guardless': cast_symbool_to_symint_guardless,
+    'ITE': _if_then_else,
```

Comment:
Naively, I expected to see some place where we have some handling that generates ITE in the code here, but I do not see any?

Reply:
you probably want to define a function like `sym_ite` to do this

Reply:
> I expected to see some place where we have some handling that generates ITE in the code here

It's used in ConvertIntSource.name(). SHAPE_ENV guard will use it to generate the guard code for the SymBool.

> you probably want to define a function like sym_ite to do this

What does this sym_ite looks like? Currently, ITE will specialize the SymBool becuase we want to propagate the guard to outer shape_env. We also have the HOP cast_symbool_to_symint uses sympy.ITE to create a a symint based on symbool and is guardless...